subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -432,24 +662,48 @@ static ssize_t write_account_key(struct bt_conn *conn,
 	return res;
 }
 
-BT_GATT_SERVICE_DEFINE(fast_pair_svc,
-BT_GATT_PRIMARY_SERVICE(BT_UUID_FAST_PAIR),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_MODEL_ID,
-		BT_GATT_CHRC_READ,
-		BT_GATT_PERM_READ,
-		read_model_id, NULL, NULL),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_KEY_BASED_PAIRING,
-		BT_GATT_CHRC_WRITE | BT_GATT_CHRC_NOTIFY,
-		BT_GATT_PERM_WRITE,
-		NULL, write_key_based_pairing, NULL),
-	BT_GATT_CCC(NULL, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_PASSKEY,
-		BT_GATT_CHRC_WRITE | BT_GATT_CHRC_NOTIFY,
-		BT_GATT_PERM_WRITE,
-		NULL, write_passkey, NULL),
-	BT_GATT_CCC(NULL, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_ACCOUNT_KEY,
-		BT_GATT_CHRC_WRITE,
-		BT_GATT_PERM_WRITE,
-		NULL, write_account_key, NULL),
-);
+static ssize_t write_additional_data(struct bt_conn *conn,
+				     const struct bt_gatt_attr *attr,
+				     const void *buf,
+				     uint16_t len, uint16_t offset, uint8_t flags)
+{
+	/* The only expected Additional Data write is with Personalized Name. fp_keys module will
+	 * return an error if it receives Personalized Name store request in inappropriate state.
+	 */
+	uint8_t decoded_data[len - FP_CRYPTO_ADDITIONAL_DATA_HEADER_LEN + sizeof(char)];
```

Comment:
Consider introducing a variable for `len - FP_CRYPTO_ADDITIONAL_DATA_HEADER_LEN` (maybe `data_len`).